File: Javascript/SearchInABinarySearchTree.js
//Leetcode Problem - 700

function TreeNode(val,right,left){
    this.val = val
    this.left = left
    this.right = right
}

function searchBST(root,val){
    if (root === null){
        return null
    }

    if (root.val === val){
        return root
    }else if(val < root.val){
        return searchBST(root.left,val)
    }else{
        return searchBST(root.right,val)
    }
}

File: Python/SearchInABinarySearchTree.py
## Leetcode Problem - 700

class TreeNode:
    def __init__(self,val=0, left=None, right=None) -> None:
        self.val = val
        self.right = right
        self.left = left


class Solution:
    def searchBST(self,root,val):
        if root == None:
            return None
        if root.val == val:
            return root
        elif val<root.val:
            leftChild = root.left
            return self.searchBST(self,leftChild,val)
        else:
            rightChild = root.right
            return self.searchBST(self,rightChild,val)
